### DISCUSSION TITLE
Is there any message area /zone in Admin theme?

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#7334] [Q&A] [answered]
I have custom module with admin ui where Click on the btn doing "ping" test to some ip. 
I would like to show msg if this "ping" success or fail. (I get bool answer from method)
Is there in Admin theme anything for that or I need to add to my own page.

**Comment by Skrypt** [comment, marked as answer]
`@await RenderSectionAsync("Messages", required: false)` in the Layout.cshtml renders the messages from an INotifier.
But that's from server side. If you want to do the same on client side you can append it in the same zone using the same HTML bootstrap elements.

https://getbootstrap.com/docs/4.0/components/alerts/

**Reply by vitalybrandes** [reply]
My page is in Admin, (default admin theme), do I need to add this section to my view?

**Reply by Skrypt** [reply]
Does your method triggered from server side or client side? Does it call a Controller / Action on server side?

**Reply by vitalybrandes** [reply]
I created admin controller with admin view  for some module.
Click on btn in that view trigger action in same admincontroller

**Reply by Skrypt** [reply]
Here is a usage example of the INotifier interface : 

``` C#
namespace OrchardCore.YourModule.Controllers
{
    public class AdminController : Controller
    {
        private readonly INotifier _notifier;

        public AdminController(INotifier notifier)
        {
            _notifier = notifier;
        }

        public YourAction()
        {
            if (something == true)
            {
                _notifier.Success("Successfully done something!");
                _notifier.Warning("Something needs to be done first before it works. Validation error...");
                _notifier.Error("Something went wrong.");
                _notifier.Information("Something finished processing.");
            }
        }
}
```

**Reply by vitalybrandes** [reply]
Perfect, Thanks!!!

P.S : 

private readonly IHtmlLocalizer H;     //required
 _notifier.Information(H["Message"])
